projects/exui-common-lib/src/lib/components/service-messages/service-messages.component.ts

```diff
@@ -0,0 +1,51 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { FeatureToggleService } from '../../services';
+
+export interface ServiceMessages {
+  [key: string]: string;
+}
+
+@Component({
+  selector: 'xuilib-service-messages',
+  templateUrl: './service-messages.component.html'
+})
+export class ServiceMessagesComponent implements OnInit {
+  public hiddenBanners: string[];
+  public filteredMessages = new Map<string, string>();
+  
+  @Input() public userRoles: string[];
+  @Input() public featureToggleKey: string;
+  @Input() public serviceMessageCookie: string;
+  
+  constructor(private featureToggleService: FeatureToggleService) { }
+
+  ngOnInit() { 
+    this.getServiceMessages();
+  }
+
+  public getServiceMessages(): void {
+    this.featureToggleService.getValue<ServiceMessages>(this.featureToggleKey, null)
+      .subscribe(messages => {
+        if (!!messages) {
+          this.createFilteredMessages(messages);        
+        }
+      });
+  }
+
+  private createFilteredMessages(messages: ServiceMessages): void {
+    this.filteredMessages.clear;
```

Comment:
Isn't this supposed to be a function call?

Reply:
I'll remove the line as it is not needed. Think it goes back to when I was deciding on the type for the filteredMessages property.